tests/queries/test_qs_combinators.py

```diff
@@ -23,6 +23,27 @@ def test_simple_union(self):
         qs3 = Number.objects.filter(num=5)
         self.assertNumbersEqual(qs1.union(qs2, qs3), [0, 1, 5, 8, 9], ordered=False)
 
+    def test_union_two_evaluations(self):
+        qs = (
+            Number.objects.filter(pk__in=[1, 2])
+            .union(Number.objects.filter(pk__in=[4, 5]))
+        )
+        self.assertEqual(list(qs.order_by('num').values_list('pk', flat=True)), [1, 2, 4, 5])
+        self.assertEqual(list(qs.order_by('-num').values_list('pk', flat=True)), [5, 4, 2, 1])
+        self.assertEqual(list(qs.order_by('-id').values_list('pk', flat=True)), [5, 4, 2, 1])
+
+        item1 = Number.objects.get(pk=1)
+        item1.num = 11
+        item1.save()
+        self.assertEqual(list(qs.order_by('num').values_list('pk', flat=True)), [2, 4, 5, 1])
+
+        # this case doesn't work but i checked query and i saw the correct order by.
```

Comment:
I wanted to check the equality status of the num column. I don't know why but order by doesn't work for the second order by column.